hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnectionManager.java

```diff
@@ -452,7 +456,8 @@ private Boolean destroyCurrentClusterConnectionAndTryNextCluster(CandidateCluste
         return false;
     }
 
-    Connection connect(Object target, Function<Object, Connection> getOrConnectFunction) {
+    <A> Connection connect(A target, Function<A, Connection> getOrConnectFunction,
+                           Function<A, Address> addressProvider) {
```

Comment:
This param is not an address provider but an address translator

Reply:
renamed